### ISSUE TITLE
inputs with :as => :radio_buttons are being skipped in validation

### ISSUE THREAD

**Issue by vitalim** [DavyJonesLocker/client_side_validations-simple_form#20]
In a form I'm building using ClientSideValidations + SimpleForm (with the gem plugin) the client side validation works for all fields except the :radio_buttons. The HTML that is produced for this group is similar to the other working ones:

`<div class="control-group radio_buttons required">
...
</div>`

Yet when the submit button is clicked, all the other inputs get the 'error' class but the radio_buttons input does not. If I remove the `:as => :radio_buttons`, the client side validations works just fine.

Any idea what is causing this? Thanks!

**Comment by AlecRust** [comment]
+1

**Comment by manishie** [comment]
It seems as though even though radio buttons are not processed, the gem is still sending data about the radio button validations to the browser.  Type 'window.ClientSideValidations.forms' into the JS console and you'll see what gets sent down.  So I wrote a bit of jQuery that checks all of my forms for radio buttons, then sees if there are any presence validations for those buttons.  If so, it displays an error message.  Note that in Rails, you'll need to wrap your related radio buttons into a div.  Hope it helps someone.  http://pastebin.com/FMzS9dBj

**Comment by tagliala** [comment]
@manishie thanks. I'm not confident to add this at the moment but this will definitely help.

**Comment by manishie** [comment]
oh, I didn't mean for you to add it!  It's just a quick hack I threw together to get around the problem.  Thanks!

**Comment by tagliala** [comment]
@manishie sure I got you didn't mean that. I just was informing everyone that I will not do this soon. PR with proper tests are welcomed

**Comment by August7752** [comment]
had this fixed?

**Comment by cooljl31** [comment]
+1

**Comment by tagliala** [comment]
@August7752 sorry for the very late reply, I've missed this notification.

This feature is not implemented and I will probably not implement this. PR with proper tests are very welcomed

@cooljl31 please use [GitHub reactions](https://github.com/blog/2119-add-reactions-to-pull-requests-issues-and-comments) to add upvotes. Other +1s are here just because there weren't github reactions at the time

**Comment by ESouza** [comment]
+1

**Comment by sekrett** [comment]
@manishie I did some experimentation and realised, that not all data is sent from server. For example radio buttons and check boxes are not present in form data attribute, regardless what validations (standard or custom) I add to them. Maybe it worked on an older version of `client-side-validations`?

**Comment by jhovad** [comment]
+1 please

**Comment by MichalRemis** [comment]
Yes validations for radio_buttons etc.. aren't in CSV Hash, because `SimpleForm` doesn't use (anymore) `FormBuilder's` methods (which CSV overrides to build validation hash) to to build these special inputs. But it uses `ActionView:Helpers::Tags` for these now, so there is some more overriding needed. I am looking into it.. wan't to make it work.

**Comment by MichalRemis** [comment]
I implemented missing validations into CSV SimpleForm (checkboxes, radios, date_time selects). Meanwhile, until it will be merged into main gem you may use it from my repo.

`gem 'client_side_validations-simple_form', github: 'MichalRemis/client_side_validations-simple_form'
`

It works only when JS is added via sprockets 
```
//= require rails.validations
//= require rails.validations.simple_form.bootstrap4
```
because I don't provide npm package.

**Comment by tagliala** [comment]
> because I don't provide npm package.

it is possible to point npm packages to github repos: 

In `package.json`:
```json
    "package": "user/repo#branch",
```
